### ISSUE TITLE
Increase size of "file" field in General Tab of Entry-Editor

### ISSUE THREAD

**Issue by ThiloteE** [JabRef/jabref#8823]
**Problem:**
1. Have multiple files attached to an entry
2. Try to select and remove an attached file from an entry

**What happens:**

Because the size of the field is so small, it has very hard to select a specific file (or specifc fileS), because the scrollbar often jumps too far or not far enough. Also, when scrolling over the end of the file field, the scrollbar of the general tab will get activated and the user will henceforth scroll within the general tab instead of the file field.

**Ideas for Solution:**
- Slightly increase size of "file" field in General Tab
- Size of General Tab should reflect the amount of files that are attached to an entry. More files --> greater size. No or only one file is attached --> small size.

**Additional context**

![grafik](https://user-images.githubusercontent.com/73715071/169037630-39bae7f3-f2b3-4edf-bda2-549405e0b8e1.png)

**Comment by AEgit** [comment]
Note, this issue has been reported before:

https://github.com/JabRef/jabref/issues/5105#issuecomment-509132201

Version 3.8.2 showed many more files than the current version (see screenshots in the linked post).

**Comment by MVP-D77** [comment]
Hello, @[[ThiloteE](https://github.com/ThiloteE)] , our group are interested in your problem and would like to work on this issue. Can we have a try?
——SE 2022 group haha, SUSTech

**Comment by MVP-D77** [comment]
And for result, simply show!
![G7XGBAA`YQYNEG{H3V%5$6C](https://user-images.githubusercontent.com/74488290/170689766-c5c469bd-3588-4584-970d-ef83f631da52.JPG)
![4`_022PM)2R69)ZBHX1 U)T](https://user-images.githubusercontent.com/74488290/170689768-5f159bcd-3691-49c1-bbe3-dc1510e3846e.JPG)

**Comment by ThiloteE** [comment]
hey, thanks yes this seems to be a lot less controversial issue than the other one :-) Thanks for the PR, I am going to try it :-)

**Comment by MVP-D77** [comment]
Thanks, but also, I just take a example, not complete!

**Comment by AEgit** [comment]
JabRef 5.8--2022-09-03--416ba16
Windows 10 10.0 amd64 
Java 18.0.2.1 
JavaFX 18.0.2+2

Hmm, this does not appear to be (fully?) solved in the current dev version. There is just more white space below the linked files (see attached picture), but there are still only 3 linked files shown, despite the fact, that there are 5 linked files. Note, that the old version of JabRef could easily handle ca. 10 linked files or so (see https://github.com/JabRef/jabref/issues/5105#issuecomment-509132201)

Behaviour in current dev version:
![JabRef58_shownfiles](https://user-images.githubusercontent.com/12767489/188314962-9b0251f2-7c40-4ef1-b370-f0998089fb5d.png)

**Comment by AEgit** [comment]
@Siedlerchr Can you re-open this issue please? As far as I can tell, the problem has not been solved yet.

**Comment by Siedlerchr** [comment]
Yes, it still has a fixed size. We are currently investigating potential ways to improve the height calculation, but seems not that easy.

**Comment by ThiloteE** [comment]
To reiterate: There is still the problem that when there are more than 3 files attached, some files will be hidden from view.

Here is where the relevant files and where most of the code that is responsible for this behaviour lives:

JabRef/src/main/java/org/jabref/gui/fieldeditors/LinkedFilesEditor.fxml
JabRef/src/main/java/org/jabref.gui/fieldeditors/LinkedFilesEditor.java
JabRef/src/main/java/org/jabref/gui/fieldeditors/LinkedFilesEditorViewModel.java
JabRef/src/main/java/org/jabref/gui/entryeditor/FieldsEditorTab.java

I used [scenic-view](https://github.com/JonathanGiles/scenic-view) to have a look at what exactly is there in the UI and what boxes and objects compete for space. I tracked it down to  the **JFXListViewSkin.class**

com.jfoenix.skins.JFXListViewSkin

**Comment by ThiloteE** [comment]
Reminder: revisit this issue once gemsfx or some other library has replaced jfoenix.

**Comment by koppor** [comment]
For historical reference - here a screenshot of JabRef 3.8.2: https://github.com/JabRef/jabref/issues/5105#issuecomment-509132201

**Comment by ilippert** [comment]
> Reminder: revisit this issue once gemsfx or some other library has replaced jfoenix.

is there by chance any expected timeline? or a link to one?
